pkgs/by-name/pf/pflotran/package.nix

```diff
@@ -0,0 +1,68 @@
+{
+  stdenv,
+  lib,
+  fetchFromBitbucket,
+  gfortran,
+  mpi,
+  petsc,
+  blas,
+  lapack,
+  parmetis,
+  hdf5,
+  mpiCheckPhaseHook,
+  python3
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "PFLOTRAN";
+  version = "6.0.1";
+
+  src = fetchFromBitbucket {
+    owner = "pflotran";
+    repo = "pflotran";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-AZXzay6GWbnxONB8Slg8gV0KN1CxGCXbJ45ZeWL1034=";
+  };
+
+  patches = [ ./make.patch ];
+
+  nativeBuildInputs = [ gfortran ];
+
+  buildInputs = [
+    petsc
+    blas
+    lapack
+    hdf5
+    parmetis
+  ];
+
+  propagatedBuildInputs = [ mpi ];
+  propagatedUserEnvPkgs = [ mpi ];
+  passthru = { inherit mpi; };
+
+  enableParallelBuilding = true;
+
+  /*
+  Pflotran does not use a "real" autotools configure script, but a simple bash
+  script, that is merely named configure. Consequently, many common mechanism
+  don't work. Thus, we need to help make to figure out some include and library
+  paths.
+  */
+  preConfigure = ''
+    substituteInPlace src/pflotran/makefile \
+      --subst-var-by "HDF5_FORTRAN_LIBS" "${lib.getLib hdf5}/lib" \
+      --subst-var-by "HDF5_FORTRAN_INCLUDE" "${lib.getDev hdf5}/include"
```

Comment:
Are the -L and -I flags are required here? The compiler wrapper should take care of that.

Reply:
Oh I wish they did. But the HDF5 linking in the Makefile is incredibly broken. They are required here.